library/src/main/kotlin/io/beyondwords/player/MediaButtonReceiver.kt

```diff
@@ -30,7 +30,7 @@ open class MediaButtonReceiver constructor(private val mediaSessionId: Int) :
             intent.putExtra(Intent.EXTRA_KEY_EVENT, KeyEvent(KeyEvent.ACTION_DOWN, keyCode))
             intent.putExtra(EXTRA_MEDIA_SESSION_ID, mediaSessionId)
             val requestCode = mediaSessionId * 1000 + keyCode // keyCode is a 3 digit number
-            val flags = if (Build.VERSION.SDK_INT >= 31) PendingIntent.FLAG_MUTABLE else 0
+            val flags = if (Build.VERSION.SDK_INT >= 31) PendingIntent.FLAG_IMMUTABLE else 0
```

Comment:
```suggestion
            val flags = when {
                Build.VERSION.SDK_INT >= 34 -> PendingIntent.FLAG_MUTABLE or PendingIntent.FLAG_ALLOW_UNSAFE_IMPLICIT_INTENT
                Build.VERSION.SDK_INT >= 31 -> PendingIntent.FLAG_MUTABLE
                else -> 0
            }
```
I saw this as a solution in a few other open source project and i think there is a reason for the PendingIntent to be mutable.
It seems like android media library does not support Android 14 yet and crashes with the same error, but as soon as they fix this issue i will adopt their solution.

Reply:
Which crash do you mean in detail? I'm a little surprised because we're already using the library successfully (with the changes) on api level 34.

Reply:
Our MediaButtonReceiver class is based of [androidx.media.session.MediaButtonReceiver](https://cs.android.com/androidx/platform/frameworks/support/+/androidx-main:media/media/src/main/java/androidx/media/session/MediaButtonReceiver.java;l=322-323), which uses FLAG_MUTABLE explicitly.
If you use androidx.media.session.MediaButtonReceiver in your app you will get the same error, since it does not check for API Level 34 yet.
I would assume this MUTABLE flag is needed in some cases by the NotificationCompat.Builder or some other parts of the system, but i cannot say for sure which, it might be fine with FLAG_IMMUTABLE too.
For the time being the safest approach would be to use `PendingIntent.FLAG_MUTABLE or PendingIntent.FLAG_ALLOW_UNSAFE_IMPLICIT_INTENT` instead of `PendingIntent.FLAG_IMMUTABLE`
If a different patch gets adopted by androidx i will make sure that we use the same PendingIntent flags as them.